doc/decisions/0012-repeating-answer-options.md

```diff
@@ -0,0 +1,259 @@
+# 12. Dynamically creating answers based on a list
+
+## Context
+
+We need the ability to dynamically generate answers based on a list item.
+
+We need to support dynamic answers that are driven by:
+
+- List collectors
+- Arbitrary functions that return a sequence of values
+- Checkbox answers
+
+Additional requirements:
+
+- Dynamic answers need to be identifiable for downstream processing
+- Dynamic answers need to be able to support user defined answer codes
+
+## Proposal
+
+Introduce a new `dynamic_answers` object that will handle the generation of any repeating answers based on a list of items.
+
+## Properties and uses:
+
+### `dynamic_answers`
+
+- A new property within the `question` block that supports answers based on a list of items, answers, or arbitrary functions 
+  that return a sequence of values.
+- `dynamic_answers` will contain two top level properties, `values` and `answers`.
+- `dynamic_answers` can be used alongside the existing static `answers` property, allowing for static and dynamic answers 
+  on the same page.
+- `dynamic_answers` must have at least one but can contain multiple `answers` to be iterated over for each literal list item.
+- `dynamic_answers` will be limited to a single block i.e. they will only be displayed on one page.
```

Comment:
Just to clarify my understanding:
- A question can have only one `dynamic_answers` object
- A all dynamic answers of a `dynamic_answers` object must be displayed within the same block
- A schema can have multiple `dynamic_answers` objects, as long as there is maximum one of these per question.

?